src/Microsoft.DotNet.Tools.Compiler.Native/IntermediateCompilation/Linux/LinuxCppCompileStep.cs

```diff
@@ -70,7 +70,8 @@ private void InitializeArgs(NativeCompileSettings config)
             //
             // Get the directory name to ensure there are no trailing slashes as they may conflict
             // with the terminating "  we suffix to account for paths with spaces in them.
-            var ilcSdkIncPath = Path.GetDirectoryName(config.IlcSdkPath);
+            char[] charsToTrim = {'\\', '/'};
+            var ilcSdkIncPath = config.IlcSdkPath.TrimEnd(charsToTrim);
```

Comment:
I have added it for consistency across all platforms - the path will not have a trailing slash (forward or backward).